substrate/frame/revive/src/inherent_handlers.rs

```diff
@@ -0,0 +1,60 @@
+use alloc::vec::Vec;
+use impl_trait_for_tuples::impl_for_tuples;
+use sp_runtime::{DispatchError, DispatchResult};
+use crate::InherentHandlerMessage;
+
+/// Defines the interface for a pallet capable of handling a specific type of inherent system message.
+///
+/// Pallets implementing this trait register themselves as potential handlers for messages
+/// identified by a unique `name`.
+pub trait InherentHandler {
```

Comment:
Maybe, could we find a different naming around `InherentHandler`? 

Just to avoid confusion with existing `Inherent` concept - extrinsics that are inherently added to each block. (InherentData, InherentIdentifier, ProvideInherent, ...)